src/lib.rs

```diff
@@ -87,11 +87,12 @@ struct Rule {
 impl Rule {
     /// Gets all byte indices in the text at which this rule matches.
     /// Contrary to the SRX 2.0 spec this does not find overlapping matches.
-    fn match_indices<'a>(&'a self, text: &'a str) -> impl Iterator<Item = usize> + 'a {
-        self.regex.captures_iter(text).map(|x| {
-            x.get(1)
-                .expect("rule regex has one capture group denoting the `after_break` part")
-                .start()
+    fn match_indices<'a>(&'a self, text: &'a str) -> impl Iterator<Item = Range<usize>> + 'a {
+        self.regex.captures_iter(text).filter_map(|x| {
+            // generally it is guaranteed that a regex has
+            // at least one match, but be lenient about
+            // errors in the srx xml files and drop those without
+            x.get(1).map(|x| x.range())
```

Comment:
I'm not sure if you've seen my previous comment about this, somehow GH had problems at that time. Anyway: the regex *must* have one capture group, otherwise it is a bug in the crate. 

I will change this to use a named group so that rules which contain capture groups themselves don't fail. We can leave this as it is for this PR but I will change it back to an `expect` when switching to a named group.

Also: What's the rationale behind changing this function to return a `Range` instead of the index at which to split? The start of the range is of course relevant but the end is just a detail that the caller should never take into account.